model/infinispan/src/main/java/org/keycloak/jgroups/impl/JGroupsJdbcPingStackConfigurator.java

```diff
@@ -63,7 +58,7 @@ public void configure(ConfigurationBuilderHolder holder, KeycloakSession session
         Supplier<DataSource> dataSourceSupplier = Arc.container().select(AgroalDataSource.class)::get;
```

Comment:
Using `Arc` will only work in Quarkus. 

To be on the safe side we could have an SPI for this one to provide a datasource. When running Quarkus, we would use `Arc`, if not, we would construct it manually. We have a similar approve for the JpaConnection with `DefaultJpaConnectionProviderFactory`. 

There are also examples on how to retrieve this from an entity manger, see https://stackoverflow.com/questions/12423930/how-to-retrieve-the-datasource-used-by-a-persistence-unit-programmatically

We could add a method to JpaConnectionProvider to return a supplier for the datasource (or the datasource). 

Let me know what you think.